frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -978,6 +1008,38 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Used to delegate a new destination to earn the stake rewards.
+        /// Possible situations:
+        /// 1 - Staker is the delegate and want to delegate someone else;
+        /// 2 - Delegate is not the staker and want to delegate someone else;
+        /// 3 - Staker is not the delegate but want to be.
+        /// Main dificulty here is because is not possible at first to know if "origin" is the staker, the delegate or someone else.
+        #[pallet::weight(T::DbWeight::get().reads_writes(1, 1))]
+        pub fn set_new_delegate(
+            origin: OriginFor<T>,
+            staker: T::AccountId,
+            contract_id: T::SmartContract,
+            destination_account: T::AccountId
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
```

Comment:
Please explain who is: `origin`, `staker` and `destination_account`.

Reply:
**origin**: The OriginFor (after the AccountId) of the user that is calling this callable function (can be the staker, the delegate, or someone else)
**staker**: The AccountId of the staker who is bonding and staking tokens on a contract "contract_id" 
**destination_account**: The AccountId of the new delegate to whom the staker wants to delegate their stake (this parameter name is not good, going go change it)